cmd/minishift/cmd/root.go

```diff
@@ -92,6 +92,8 @@ var RootCmd = &cobra.Command{
 		// If profile name is 'minishift' then ignore the vaild profile check.
 		if constants.ProfileName != constants.DefaultProfileName {
 			checkForValidProfileOrExit(cmd)
+			//create the default profile dirs for global config
+			createMinishiftDirs(state.NewMinishiftDirs(constants.GetProfileHomeDir(constants.DefaultProfileName)))
```

Comment:
This does not look correct as we have `createMinishiftDirs` before `ensureConfigFileExists(constants.GlobalConfigFile)`. Need to find out what actually happening. 

```code
// creating all directories for minishift run
createMinishiftDirs(state.InstanceDirs)

// Ensure the global viper config file exists.
ensureConfigFileExists(constants.GlobalConfigFile)
```

Reply:
The issue is that when ~/.minishift/config is not present and user runs `minishift profile set  NON_DEFAULT_PROFILE_NAME` , `createMinishiftDirs(state.InstanceDirs)` only created the directory structure in `~./minishift/profiles`  and not the `~/.minishift/config` . Hence `ensureConfigFileExists(constants.GlobalConfigFile)` gives error as it expects that `~/.minishift/config` would be present.

```
$ tree ~/.minishift
/Users/lmohanty/.minishift
├── cache
│   ├── images
│   ├── iso
│   └── oc
└── profiles
    └── Test
        ├── addons
        ├── certs
        ├── config
        ├── logs
        ├── machines
        └── tmp
```

Reply:
@LalatenduMohanty Right, that is why i am adding that line to create the directory structure on the upper level then profile, so that the config directory exists.